packages/dds/task-manager/src/taskManager.ts

```diff
@@ -455,7 +457,7 @@ export class TaskManagerClass
 			if (eventTaskId !== taskId) {
 				return;
 			}
-			if (messageId !== undefined && messageId <= currentMessageId) {
+			if (messageId !== undefined && messageId <= nextPendingMessageId) {
```

Comment:
I'm wondering if there are cases where this runs before we've submitted the volunteer op (and therefore nextPendingMessageId is undefined).  E.g.
1. Submit "abandon"
2. Disconnect before seeing the ack
3. Attempt subscribe (but since disconnected, will be waiting for connect before volunteering)
4. Reconnect, process the ack for the "abandon" while catching up